### PR TITLE
[Spring-Cloud] Update API version to 2022-01-01-preview

### PR DESCRIPTION
**Pull request by yuwzho**
---

## Description
This PR upgrades the commands who use "preview" API version to switch to the latest preview API version. Because the 2022-01-01-preview is the only API version to support the Enterprise sku. As 2022-01-01-preview API version introduces some breaking changes since last preview API version. This PR also address these breaking changes in API level, and keep consistency on command/arguments/response 

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring-Cloud

**Comment by jsntcy** [comment]
@yuwzho, please summarize what you changed in description.

File: src/spring-cloud/azext_spring_cloud/_transformers.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 
 # pylint: disable=line-too-long
+from msrestazure.tools import parse_resource_id
```
Comment (jsntcy) [inline src/spring-cloud/azext_spring_cloud/_transformers.py:8]: Use "from azure.mgmt.core.tools import parse_resource_id" instead as "msrestazure.tools" will be deprecated soon.

File: src/spring-cloud/azext_spring_cloud/vendored_sdks/appplatform/v2022_01_01_preview/models/_models_py3.py [hunk 2]
```diff
@@ -4060,6 +4062,7 @@ class JarUploadedUserSourceInfo(UploadedUserSourceInfo):
         'version': {'key': 'version', 'type': 'str'},
         'relative_path': {'key': 'relativePath', 'type': 'str'},
         'runtime_version': {'key': 'runtimeVersion', 'type': 'str'},
+        'jvm_options': {'key': 'jvmOptions', 'type': 'str'},
```
Comment (jsntcy) [inline src/spring-cloud/azext_spring_cloud/vendored_sdks/appplatform/v2022_01_01_preview/models/_models_py3.py:4065]: Is this change missing in last PR for vendored SDK update?

Comment (yuwzho) [reply src/spring-cloud/azext_spring_cloud/vendored_sdks/appplatform/v2022_01_01_preview/models/_models_py3.py:4065]: Yes, sad story, the missing is from swagger side. We have added it back

File: src/spring-cloud/azext_spring_cloud/_transformers.py (hunk 1 above)
Comment (yuwzho) [reply src/spring-cloud/azext_spring_cloud/_transformers.py]: Updated, as well as import in other files

**Review by jsntcy — APPROVED** [review]
:shipit:

**Comment by yuwzho** [comment]
#4294

**Pull request merged**